services/infrastructure/src/mongoDataStorage/MongoDataStoreClient.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System;
+using System.Linq.Expressions;
+using System.Threading.Tasks;
+using artiso.AdsdHotel.Infrastructure.DataStorage;
+using MongoDB.Driver;
+
+namespace artiso.AdsdHotel.Infrastructure.MongoDataStorage
+{
+    public class MongoDataStoreClient : IDataStoreClient
+    {
+        private readonly MongoClient client;
+        private readonly IMongoDatabase db;
+        private readonly string collection;
+
+        public MongoDataStoreClient(Uri endpoint, string database, string collection)
+        {
+            var connectionString = $"{endpoint.Scheme}://{endpoint.Host}:{endpoint.Port}";
+            client = new MongoClient(connectionString);
+            db = client.GetDatabase(database);
+            this.collection = collection;
+
+        }
+
+        public async Task AddOrUpdate<T>(T entity)
+        {
+            var col = db.GetCollection<T>(collection);
+            //var filter = Builders<GuestInformationRecord>.Filter.Eq(x => x.OrderId, message.OrderId);
+            //var update = Builders<GuestInformationRecord>.Update.Set(x => x.GuestInformation, message.GuestInformation);
+            //var options = new FindOneAndUpdateOptions<GuestInformationRecord>();
+            //options.IsUpsert = true;
+            //var proj = await collection.FindOneAndUpdateAsync(filter, update, options).ConfigureAwait(false);
+            await col.InsertOneAsync(entity).ConfigureAwait(false);
```

Comment:
does insertOneAsync work as add or Update?

Reply:
I have to try that out. I opened another issue for updating entries.